services/frontend-service/src/ui/components/SelectionDialog/GenericSelectionDialog.tsx

```diff
@@ -0,0 +1,83 @@
+/*This file is part of kuberpult.
+
+Kuberpult is free software: you can redistribute it and/or modify
+it under the terms of the Expat(MIT) License as published by
+the Free Software Foundation.
+
+Kuberpult is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+MIT License for more details.
+
+You should have received a copy of the MIT License
+along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>.
+
+Copyright freiheit.com*/
+import * as React from 'react';
+import { Checkbox } from '../dropdown/checkbox';
+import { ConfirmationDialog } from '../dialog/ConfirmationDialog';
+
+export type GenericSelectionDialogProps = {
+    selectables: string[];
+    onSubmit: () => void;
+    onCancel: () => void;
+    open: boolean;
+    multiSelect: boolean;
+    headerLabel: string;
+    confirmLabel: string;
+    onEmptyLabel: string;
+    selectedItems: string[];
+    setSelectedItems: React.Dispatch<React.SetStateAction<string[]>>;
+};
+
+export const GenericSelectionDialog: React.FC<GenericSelectionDialogProps> = (props) => {
+    const selectedSelectables = props.selectedItems;
+    const setSelectedSelectables = props.setSelectedItems;
```

Comment:
nitpick: this could be phrased simpler:
```suggestion
    const { selectedSelectables, setSelectedSelectables }  = props;
```